File: src/java/EntityFX/src/EntityFX/Core/Linpack/LinpackBenchmark.java
package EntityFX.Core.Linpack;

import java.io.FileNotFoundException;
import java.io.IOException;

import EntityFX.Core.Writer;
import EntityFX.Core.Generic.BenchResult;
import EntityFX.Core.Generic.BenchmarkBase;

public class LinpackBenchmark extends BenchmarkBase<LinpackResult> {

    private Linpack linpack;

    public LinpackBenchmark(final Writer writer, final boolean printToConsole) throws FileNotFoundException {
        super(writer, printToConsole);
        linpack = new Linpack(printToConsole);
        Ratio = 10;
    }

    @Override
    public LinpackResult benchImplementation() throws IOException {
        return linpack.run_benchmark(2000);
    }

    /* error value type 'BenchResult' of method PopulateResult */
    @Override
    public BenchResult populateResult(final BenchResult benchResult, final LinpackResult linpackResult) {
        benchResult.Result = linpackResult.MFLOPS;
        benchResult.Points = linpackResult.MFLOPS * this.Ratio;
        benchResult.Units = "MFLOPS";
        benchResult.Output = linpackResult.output;
        return benchResult;
    }

    @Override
    public void warmup(final Double aspect) {
    }
}

File: src/python/entityfx/linpack_benchmark.py
from entityfx.linpack import Linpack
from entityfx.benchmark_base import BenchmarkBase
from entityfx.writer import Writer

class LinpackBenchmark(BenchmarkBase):
    
    def __init__(self, writer: Writer, print_to_console : bool=True) -> None:
        super().__init__(writer, print_to_console)
        self.__linpack = Linpack(True)
        self.ratio = 10
    
    def benchImplementation(self):
        return self.__linpack.bench(1000)
    
    def populateResult(self, bench_result, linpack_result):
        bench_result["Points"] = linpack_result["MFLOPS"] * self.ratio
        bench_result["Result"] = linpack_result["MFLOPS"]
        bench_result["Units"] = "MFLOPS"
        bench_result["Output"] = linpack_result["Output"]
        return bench_result
    
    def warmup(self, aspect : float=.05) -> None:
        pass